File: src/main/java/com/yupi/template/service/AIScoringService.java
package com.yupi.template.service;

import com.yupi.template.model.dto.evaluation.AIScoreResult;
import com.yupi.template.model.dto.evaluation.EvaluationResult;

/**
 * AI评分服务接口
 *
 * @author <a href="https://codefather.cn">编程导航学习圈</a>
 */
public interface AIScoringService {

    /**
     * 对模型回答进行AI评分（单评委）
     *
     * @param question 问题
     * @param modelResponse 模型回答
     * @return 评分结果
     */
    EvaluationResult score(String question, String modelResponse);

    /**
     * 对模型回答进行多评委交叉验证评分
     *
     * @param question 问题
     * @param modelResponse 模型回答
     * @param testedModelName 被测试的模型名称（用于排除，避免自己评自己）
     * @return 多评委评分结果
     */
    AIScoreResult scoreWithMultipleJudges(String question, String modelResponse, String testedModelName);
}

File: go-backend/internal/service/ai_scoring_service.go
// Package service AI 评分服务接口定义
// @author <a href="https://codefather.cn">编程导航学习圈</a>
package service

import (
	"ai-test-go/internal/model/dto"
)

// AIScoringService AI 评分服务：单评委评分、多评委交叉验证
type AIScoringService interface {
	// Score 对模型回答进行 AI 评分（单评委）
	// question 问题，modelResponse 模型回答，userID 用于统计模型使用量
	Score(question, modelResponse string, userID int64) (*dto.EvaluationResult, error)

	// ScoreWithMultipleJudges 对模型回答进行多评委交叉验证评分
	// testedModelName 被测试的模型名称，用于排除该模型避免自己评自己
	ScoreWithMultipleJudges(question, modelResponse, testedModelName string, userID int64) (*dto.AIScoreResult, error)
}
